Vic2ToHoI4/Source/HOI4World/WarCreator/HoI4WarCreator.cpp

```diff
@@ -72,39 +72,47 @@ void HoI4WarCreator::addAllTargetsToWorldTargetMap()
 }
 
 
-void HoI4WarCreator::addTargetsToWorldTargetMap(shared_ptr<HoI4::Country> country)
+void HoI4WarCreator::addTargetsToWorldTargetMap(std::shared_ptr<HoI4::Country> country)
 {
 	if (country->getGovernmentIdeology() != "democratic")
 	{
 		int maxGCWars = 0;
-		for (auto GC: getDistancesToGreatPowers(country))
+		for (const auto& greatPowerTag: mapUtils.getGPsByDistance(*country, theWorld->getGreatPowers()))
 		{
-			if (maxGCWars < 2)
+			auto greatPower = theWorld->findCountry(greatPowerTag);
+			if (greatPower == nullptr)
 			{
-				string HowToTakeGC = HowToTakeLand(GC.second, country, 3);
-				if (HowToTakeGC == "noactionneeded" || HowToTakeGC == "factionneeded" || HowToTakeGC == "morealliesneeded")
+				continue;
+			}
+
+			if (maxGCWars > 1)
+			{
+				continue;
+			}
+
+			std::string HowToTakeGC = HowToTakeLand(greatPower, country, 3);
+			if (HowToTakeGC == "noactionneeded" || HowToTakeGC == "factionneeded" || HowToTakeGC == "morealliesneeded")
+			{
+				if (greatPower != country)
 				{
-					if (GC.second != country)
+					auto relations = country->getRelations(greatPower->getTag());
+					if ((relations) && (relations->getRelations() < 0))
```

Comment:
The double parentheses aren't needed here, but maybe it's your preference.

Reply:
I'll address in future cleanup efforts.